api/src/main/java/bio/terra/tanagra/common/Paginator.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.tanagra.common;
+
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.tanagra.proto.utils.pagination.PageSizeOffset;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.util.Base64;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/** Implements simple pagination for sorted lists in memory. */
+public final class Paginator<T> {
+  private Paginator() {}
+
+  /**
+   * Returns a {@link Page} for the results.
+   *
+   * @param allResults the sorted list of all results being paginated.
+   * @param pageSize the maximum number of results to have per page.
+   * @param pageToken the token of the page to retrieve. If null or empty, the first page is
+   *     retrieved.
+   */
+  public static <T> Page<T> getPage(List<T> allResults, int pageSize, @Nullable String pageToken) {
+    Preconditions.checkArgument(pageSize > 0, "pageSize must be greater than 0 to paginate.");
+    if (pageToken == null || pageToken.isEmpty()) {
+      return getPageInternal(
+          allResults, PageSizeOffset.newBuilder().setOffest(0).setPageSize(pageSize).build());
+    }
+    PageSizeOffset pageSizeOffset = decode(pageToken);
+    if (pageSizeOffset.getPageSize() != pageSize) {
+      throw new BadRequestException(
+          String.format(
+              "Token page size '%d' does not match input page size '%d'.",
+              pageSizeOffset.getPageSize(), pageSize));
```

Comment:
Technically you could allow this in the API, though it would depend on your implementation (this current implementation cannot support page size changes mid-stream). See other comment about `next_id`